include/picongpu/fields/FieldE.tpp

```diff
@@ -178,98 +178,100 @@ void FieldE::laserManipulation( uint32_t currentStep )
      * in time
      */
     constexpr float_X laserTimeShift = laser::initPlaneY * CELL_HEIGHT / SPEED_OF_LIGHT;
+
+    const uint32_t numSlides = MovingWindow::getInstance().getSlideCounter(currentStep);
+
     /* Disable laser if
      * - init time of laser is over or
      * - we have periodic boundaries in Y direction or
      * - we already performed a slide
      */
-    if(
-        laserProfile::INIT_TIME == float_X(0.0) || /* laser is disabled e.g. laserNone */
-        ( currentStep * DELTA_T  - laserTimeShift ) >= laserProfile::INIT_TIME ||
-        Environment<simDim>::get().GridController().getCommunicationMask( ).isSet( TOP )
-    )
-    {
-        return;
-    }
-    const uint32_t numSlides = MovingWindow::getInstance().getSlideCounter(currentStep);
-    if( numSlides != 0 )
-    {
-        return;
-    }
-    else
+    bool const laserNone = ( laserProfile::INIT_TIME == float_X(0.0) );
+    bool const laserInitTimeOver =
+        ( ( currentStep * DELTA_T  - laserTimeShift ) >= laserProfile::INIT_TIME );
+    bool const topBoundariesArePeriodic =
+        ( Environment<simDim>::get().GridController().getCommunicationMask( ).isSet( TOP ) );
+    bool const boxHasSlided = ( numSlides != 0 );
+
+    bool const disableLaser =
```

Comment:
Note: the double negation is a little bit hard to read. Optional, negate all conditions above and call this variable `enableLaser`.

Reply:
offline discussion: since this method is expected to manipulate the laser and each of them is a sufficient abort criteria an or upon disable criteria is ok here.
the `enableLaser` as `!disableLaser` is still a bit funny but well